internal/pkg/aws/acm/acm.go

```diff
@@ -24,7 +38,58 @@ func New(s *session.Session) *ACM {
 	}
 }
 
-// ValidateCertAliases is a no-op and is supposed to validates if aliases are all valid against the provided ACM certificates.
+// ValidateCertAliases validates if aliases are all valid against the provided ACM certificates.
 func (a *ACM) ValidateCertAliases(aliases []string, certs []string) error {
+	validAliases := make(map[string]bool)
+	ctx, cancelWait := context.WithTimeout(context.Background(), waitForFindValidAliasesTimeout)
+	defer cancelWait()
+	g, _ := errgroup.WithContext(ctx)
+	var mux sync.Mutex
+	for i := range certs {
+		cert := certs[i]
+		g.Go(func() error {
+			validCertAliases, err := a.findValidAliasesAgainstCert(aliases, cert)
+			if err != nil {
+				return err
+			}
+			mux.Lock()
+			defer mux.Unlock()
+			for _, alias := range validCertAliases {
+				validAliases[alias] = true
+			}
+			return nil
+		})
+	}
+	if err := g.Wait(); err != nil {
+		return err
+	}
+	for _, alias := range aliases {
+		if !validAliases[alias] {
+			return fmt.Errorf("%s is not a valid domain against %s", alias, strings.Join(certs, ","))
+		}
+	}
 	return nil
 }
+
+func (a *ACM) findValidAliasesAgainstCert(aliases []string, cert string) ([]string, error) {
+	resp, err := a.client.DescribeCertificate(&acm.DescribeCertificateInput{
```

Comment:
We can use `DescribeCertificateWithContext` and pass the group's context so that if the context is canceled we don't wait for the response